Route status updates

dtaylor113 opened the pull request:
Route status styles should be consistent with node status
https://jira.coreos.com/browse/CONSOLE-517
**Before:**
![image](https://user-images.githubusercontent.com/12733153/41313897-2583006e-6e59-11e8-9341-6da7223d17e8.png)
**After:**
![image](https://user-images.githubusercontent.com/12733153/41313927-3ba98516-6e59-11e8-9d83-e8eabf2ff93b.png)


Route Details: Add Error Message box around Status Warnings
https://jira.coreos.com/browse/CONSOLE-533
**Before:**
![image](https://user-images.githubusercontent.com/12733153/41313947-4ac87390-6e59-11e8-89d2-9d70bdda2837.png)
**After:**
![image](https://user-images.githubusercontent.com/12733153/41313950-4e15f95a-6e59-11e8-95d5-d8a524d98caf.png)

@openshift/team-ux-review

benjaminapetersen commented on frontend/public/style/_icons.scss:
```diff
@@ -6,6 +6,14 @@
   color: $color-red-error;
 }
 
+.route-accepted {
```
Is the `$color-green-success` different than the color `text-success` provides?   And if so, there isn't an existing `text-` class that provides this color?  Just curious if we need special `route-` classes for the color change.

benjaminapetersen replied:
Meaning if it is supposed to be consistent with `node`, per the JIRA, perhaps a generic class helps ensure future consistency if the pattern continues elsewhere.

dtaylor113 replied:
Hi, updated so route & node are using same css, pls see screenshots below.  -thanks

spadgett commented:
cc @tlwu2013

spadgett commented on frontend/public/style/_icons.scss:
```diff
@@ -6,6 +6,14 @@
+
+.route-rejected {
+  color: $color-red-error;
+}
```
```css
.node-ready,
.route-accepted {
  color: $color-green-success;
}

.node-not-ready,
.route-rejected  {
   color: $color-red-error;
 }
```

spadgett commented on frontend/public/components/_route.scss:
```diff
@@ -4,6 +4,11 @@
+  border: solid 2px $red;
+  color: $color-red-error;
+  font-weight: 700;
+  padding: 6px;
```
Is an existing class with this style we can use?

@tlwu2013 Just confirming this is the error style you had recommended (see screenshots).

dtaylor113 replied:
There is something in 'Applications -> Cluster Service Versions -> Overview' but I couldn't find the exact code.

dtaylor113 replied:
> is the green "Ready" using PF green 400? I think it should be per #123.

hi, yes, I'm going to change to use pf green 400

>Is the error text intentionally bold when inside the red area?

yes, font-weight 700, probably shouldn't be.  I was trying to only bold the first line/topic sentence per @tlwu2013 mockup in JIRA.  The way it's implemented, some additional coding would be needed to isolate the first sentence.  

We might not inline the error here, looking into @spadgett's suggestion:
> We might want to use the Conditions component from #97 instead of inlining the error here. It has more detail and should have the same basic info about accepted / rejected routes.

spadgett commented:
This will likely conflict with #123, which tweaks the color.

beanh66 commented:
@dtaylor113 two questions, (1) is the green "Ready" using PF green 400? I think it should be per #123. (2) Is the error text intentionally bold when inside the red area?

dtaylor113 commented:
Hi @beanh66 

> is the green "Ready" using PF green 400? I think it should be per #123.

yes, I'm going to change to use pf green 400

>Is the error text intentionally bold when inside the red area?

yes, font-weight 700, probably shouldn't be.  I was trying to only bold the first line/topic sentence per @tlwu2013 mockup in JIRA.  The way it's implemented, some additional coding would be needed to isolate the first sentence.  

We might not inline the error here, looking into @spadgett's suggestion:
> We might want to use the Conditions component from #97 instead of inlining the error here. It has more detail and should have the same basic info about accepted / rejected routes.

dtaylor113 commented:
Hi, rebased and updated to use `$color-pf-green-400`
![image](https://user-images.githubusercontent.com/12733153/41360527-f6f66400-6efa-11e8-871c-5a2349964a21.png)

Latest error status msg box:
![image](https://user-images.githubusercontent.com/12733153/41360558-0842e04e-6efb-11e8-8293-1db9eb3c491d.png)

@spadgett, I like the Conditions component, ok to do that in a follow up PR as #97 hasn't merged yet. 
thanks

beanh66 commented:
@dtaylor113 @spadgett Do you have other examples in OpenShift of where we show these error snippets? Wondering if this should be styled as a PF inline notification error or something else.

dtaylor113 commented:
HI @beanh66, can you access https://jira.coreos.com/browse/CONSOLE-533  mockup and screenshot there

beanh66 commented:
@dtaylor113 Looks like I cannot :( 
<img width="728" alt="screen shot 2018-06-13 at 11 55 21 am" src="https://user-images.githubusercontent.com/18057044/41362963-b782c132-6f00-11e8-9aac-492fe0a771a9.png">

dtaylor113 commented:
Removed/reverted the error messages styles, will follow up with PR to add the Conditions component as in #97.   
**Latest:**
![image](https://user-images.githubusercontent.com/12733153/41368582-55a342b0-6f10-11e8-8e99-0c58043717fc.png)

spadgett commented on frontend/public/components/routes.tsx:
```diff
@@ -116,13 +116,13 @@ export const RouteStatus: React.SFC<RouteStatusProps> = ({obj: route}) => {
-      Accepted
+        <div className="fa fa-check route-accepted co-m-route-status-icon" aria-hidden="true"></div>
+        <span className="route-accepted">Accepted</span>
       </React.Fragment>;
```
Any reason not to use...? (Here and below)

```jsx
return <span className="route-accepted">
  <span className="fa fa-check co-m-route-status-icon" aria-hidden="true"></span>
  Accepted
</span>;
```

Missed this on previous review, but the icon should probably be an inline element like `<span>` or `<i>` instead of a block element  (`<div>`).

dtaylor113 replied:
> Any reason not to use...? (Here and below)

Not sure I understand your question, by ? if you mean inline if, it's not a "this or that" -it can be three values Pending, Accepted, Rejected.

spadgett replied:
Any reason not to use the code I pasted with one route-accepted class on the parent span.

dtaylor113 replied:
> ...icon should probably be an inline element like <span> or <i> instead of a block element (<div>).
Fixed, thanks

dtaylor113 commented:
> Are there reasons given for an accepted route? or just the status marker?

I believe just the status marker.  I think accepted = success, so no more explanation needed :-)

beanh66 commented:
@cshinn @dtaylor113 I agree, the most recent version is more inline with what I would expect for this use case. It felt like the red block/ error handling was too strong before.

serenamarie125 commented:
@dtaylor113 can you add a screenshot with more context?  I'm curious about the spacing around the Rejection "reason"

spadgett commented:
> can you add a screenshot with more context? I'm curious about the spacing around the Rejection "reason"

FYI, we plan to remove that text and switch to a common conditions table that is reused across other resources

spadgett commented:
See #97
